Adds the circle game

PKPenguin321 opened the pull request:
Started by using the "circle" emote.

See: https://tgstation13.org/phpBB/viewtopic.php?f=9&t=11951

Sprite:
![circled hand inhand](https://user-images.githubusercontent.com/8345184/27751797-df6558fc-5d92-11e7-8fde-8fae0e22f2e2.png)

**Not featured in this PR** but featured in that thread: On-mob sprites (couldn't think of an easy way to do them), starting the game by "using" an empty hand (it's started with an emote instead), messages telling you who examined you.

🆑
add: You can now play the circle game and prank your coworkers by using the "circle" emote.
/:cl:

PKPenguin321 committed:
adds the circle game
- adds an inventory sprite to weapons.dmi
- adds the code for the hand "object" to weaponry.dm
- adds emote datum code for making your hand a circle to emote.dm

lzimann commented on code/modules/mob/living/emote.dm:
```diff
@@ -478,3 +478,14 @@
+
+/datum/emote/living/circle/run_emote(mob/user, params)
+	. = ..()
+	var/obj/item/weapon/circlegame/N = new(user)
```
add an `if(.)` before these new things

PKPenguin321 replied:
This broke it and made it never work properly, so I removed it

PKPenguin321 replied:
Talked to leoz on IRC and the if(.) is not actually needed so please disregard this review

GunHog commented:
Seems pointless and weird. I support.

Niknakflak commented:
The game is much better than the circle.

Lame

PKPenguin321 committed:
fixes
- makes sure the emote is actually ran before creating objects
- qdels the new object if it fails to get placed into the mobs hand

PKPenguin321 committed:
adds a message to the user when emote fails

dannno commented:
This is a wrinklebrain PR.

PKPenguin321 commented:
I'm working on some tweaks to this, don't merge

PKPenguin321 committed:
added special examine reactions
- added new proc to items.dm
- added handling for the proc in human/examine.dm
- added special examine reaction to the circled hand in weaponry.dm
- removed a check on the circle emote that made it not work in emote.dm

PKPenguin321 commented:
Added a new thing that allows items to have unique actions reactions when they're equipped on a mob and the mob is examined. Also allows certain items to have unique examine text when on mobs. Circle game uses this to tell you when people examine you while you're making a circle.

PKPenguin321 commented on code/game/objects/items/weapons/weaponry.dm:
```diff
@@ -556,3 +556,18 @@
+/obj/item/weapon/circlegame/examine_reaction(mob/living/carbon/human/owner, mob/examiner)
+	to_chat(examiner, "<span class='userdanger'>You accidentally look at [owner]'s circled hand!</span>")
+	to_chat(owner, "<span class='notice'>You catch [examiner] glimpsing at your circled hand.</span>")
+	..()
```
I don't think this parent call works and I'm not sure why, help would be appreciated. Basically the special_examine_text doesn't show properly for the circled hand in particular, but the special_examine_text system works fine for everything else.

Cyberboss replied:
Should just be `return ..()` I think

Cyberboss commented on code/game/objects/items.dm:
```diff
@@ -67,6 +67,7 @@ GLOBAL_VAR_INIT(rpg_loot_items, FALSE)
 	var/origin_tech = null	//Used by R&D to determine what research bonuses it grants.
 	var/needs_permit = 0			//Used by security bots to determine if this item is safe for public use.
 
+	var/special_examine_text = null //If an object should have different description text while it's being worn on a human and the human is examined, set this.
```
Remove the ` = null`

Cyberboss commented on code/game/objects/items.dm:
```diff
@@ -371,6 +372,12 @@ GLOBAL_VAR_INIT(rpg_loot_items, FALSE)
 
+/obj/item/proc/examine_reaction(mob/living/carbon/human/owner, mob/examiner)
+	if(special_examine_text)
+		return 1
```
TRUE

Cyberboss commented on code/game/objects/items.dm:
```diff
@@ -371,6 +372,12 @@ GLOBAL_VAR_INIT(rpg_loot_items, FALSE)
+	if(special_examine_text)
+		return 1
+	else
+		return 0
```
FALSE

Cyberboss commented on code/game/objects/items/weapons/weaponry.dm:
```diff
@@ -556,3 +556,18 @@
+
+/obj/item/weapon/circlegame/examine_reaction(mob/living/carbon/human/owner, mob/examiner)
+	to_chat(examiner, "<span class='userdanger'>You accidentally look at [owner]'s circled hand!</span>")
+	to_chat(owner, "<span class='notice'>You catch [examiner] glimpsing at your circled hand.</span>")
```
I hope you realize these to_chats happen before any of the default examine text

PKPenguin321 replied:
Yeah. Not sure how to fix

octareenroon91 replied:
You could devolve more of the examine logic into `examine_reaction()`.
`/mob/living/carbon/human/examine(mob/user)`uses `to_chat()` and doesn't return, so you could break the output into multiple to_chat instances.

octareenroon91 replied:
Wait, I'm wrong. 
1. the userdanger line goes into the examine msg stream. 
2. the to-owner message goes now.

PKPenguin321 committed:
defines, a fix

PKPenguin321 committed:
undoes last two commits

PKPenguin321 commented:
Undid shitty implementation of examining stuff. Maybe some other time.

Implementation is now clean and this should be ready to go.

Xhuis commented:
I don't understand the hate going around about this. It's a harmless, fun little feature, and the framework it adds can be useful for other stuff, too. We have maintainers here _refusing_ to help, and not answering when PK offered to just add the framework.

Why get so upset over something so small?

demonfiren commented:
well remie hates it due to getting bopped as per forum post

dannno commented:
I'm disappointed in the behaviour of our maintainers towards Pkp in regards to this situation.
Come on.

Xhuis commented:
Cyberboss has been helpful in keeping the code modern and up-to-date, and I think he shouldn't be complained about here. Others are refusing to help due to a personal bias, which seems wrong to me, but then again, I'm not a maintainer. I do think that the maintainers who dislike the feature should say why and give good reasons other than their own personal dislike (while they're entitled to that, one maintainers' dislike is no reason to close a PR on subject alone.)

Cyberboss commented:
I have nothing against this PR, I'll merge it if a good solution is presented. I just don't really have the spare energy to come up with it myself

PKPenguin321 commented:
I solved it by removing the examine functionality since it was never required in the first place

Xhuis commented:
Hmmm. I would add that framework back in, but it already caused a lot of drama alone and there doesn't seem to be an easy way for it to be implemented.

@PKPenguin321 Don't, like, flaunt the fact that this got merged. I know of at least two maintainers that this is going to make upset on the mere content alone; no reason to stoke the fires more than they have to be. I do hope that they see that it's just a harmless thing that has really zero impact and is just something for people to have fun with, as that might sway them.

demonfiren commented:
Fun is illegal, though.
